src/guess_vector_type.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+"""Infer single stranded (ssAAV) vs. self-complementary (scAAV) vector type.
+
+Approach:
+- scAAV reads aligned to the vector contig almost always have two alignments (forward
+  and reverse-complement), with rare single alignments coming from contaminants and the
+  plasmid backbone.
+- ssAAV reads aligned to the vector more often have single alignments; reads with two
+  alignments most often come from "snapback" vector genomes.
+- Therefore, we calculate the fraction of reads aligned to the vector contig that have
+  exactly one alignment. If this fraction is below a threshold, we report scAAV,
+  otherwise ssAAV.
+"""
+
+from __future__ import annotations
+
+import argparse
+import collections
+import sys
+
+import pysam
+
+SC_MAX_THRESHOLD = 0.1
```

Comment:
If we have a ss vector with a lot of snapbacks, are going to guess a sc AAV vector.
We can simplify this and use the vector length (i.e 2.4KB < sc )